src/main/java/com/joyent/manta/monitor/commands/PutFileCommand.java

```diff
@@ -44,13 +45,16 @@ public boolean execute(final MantaOperationContext context) throws Exception {
 
         try {
             stopwatch.start();
+            Histogram.Timer timer = context.getRequestPutHistogramsKey()
```

Comment:
Have you checked the thread safety of the `Histogram` and `Timer` classes? I took a quick glance and it does seem to be using the thread-safe `DoubleAdder` class in the `Child` class. You will need to make sure that you do not need to lock when updating this value. This will require deeply understanding the `Histogram` implementation.

Reply:
Yes, it should be thread safe, as it is a requirement for any prometheus client library.  https://prometheus.io/docs/instrumenting/writing_clientlibs/#overall-structure